[B] Add NETHER_BRICK & QUARTZ_BLOCK to Step (Fixes BUKKIT-4074)

Peter200lx opened the pull request:
## The Issue:

Minecraft added two new slab types. (Nether Brick in MC 1.4.6 and Quartz in MC 1.5). Bukkit's Step class had not taken these into account, so setMaterial() and getMaterial() couldn't be used to set/read the those types of slabs.
## Justification for this PR:

This pull request simply adds the two new materials to the texture array for steps.
## JIRA Ticket:

BUKKIT4074 - https://bukkit.atlassian.net/browse/BUKKIT-4074
### Other Comments

This doesn't deal with the fact that double slabs also have two additional type additions (4 total new data vales), but that can't be handled quite as nicely as there is not a clean Material type that could be used.

gmcferrin commented:
@Peter200lx Yes, it should be QUARTZ_BLOCK

turt2live commented:
~~Can you provide testing materials for this please?~~

turt2live commented:
Disregard the testing materials comment... here they are:

[JAR File](https://docs.google.com/file/d/0B6JIYSBkLkAnVkRGc3h1R2VwalE/edit?usp=sharing)
[Source](http://pastebin.com/nkAW6erT)

Confirmed working with the above materials. 

```
    /*
     * To activate this PR testing simply interact with a STEP or DOUBLE_STEP.
     * This will go through the list of materials (known to work with steps) and
     * set the STEP (or DOUBLE_STEP) to that material.
     */
```

turt2live commented:
When possible please amend the commit message to match the format outlined in our [Contributing Guidelines](https://github.com/Bukkit/Bukkit/blob/master/CONTRIBUTING.md#commit-message-example).

Peter200lx committed:
Add missing materials to Step. Fixes BUKKIT-4074

When Minecraft 1.4.6 was released, Nether Brick texturing to steps
was added. Minecraft 1.5 added Quartz texturing to steps. When Bukkit
was updated to these version the textures for steps were not applied.

Currently it is not possible to set the texture of steps to quartz
or nether brick. This commit fixes that by adding the respective values
to the allowable materials list.

Peter200lx commented:
Alright, is it now formatted correctly? Please let me know if there's anything else you need tweaked.

turt2live commented:
It's not quite perfect, sadly.

An example commit would be this one: https://github.com/Bukkit/Bukkit/commit/2ccd6714a1039f58ba171438af36ffada86343e1

Your commit message should look something like this:

> Add missing materials to Step. Fixes BUKKIT-4074
> 
> When Minecraft 1.4.6 was released, Nether Brick texturing to steps was added. Minecraft 1.5 added Quartz texturing to steps. When Bukkit was updated to these version the textures for steps were not applied.
>  
> Currently it is not possible to set the texture of steps to quartz or nether brick. This commit fixes that by adding the respective values to the allowable materials list.

Peter200lx commented:
Ah, you were looking for more of a "what was missing", "how it was fixed" text, I misunderstood. Is easy enough to update.

turt2live commented:
Thanks for the update. Hopefully it gets pulled now :)

turt2live commented:
Pulled, thanks.
https://github.com/Bukkit/Bukkit/commit/b4980840ab090ad59cc754f49c8395bddee92f61
